Source/santa-driver/SantaDecisionManager.cc

```diff
@@ -19,6 +19,10 @@ OSDefineMetaClassAndStructors(SantaDecisionManager, OSObject);
 
 #pragma mark Object Lifecycle
 
+template<> uint64_t SantaCacheHasher<santa_vnode_id_t>(santa_vnode_id_t const& s) {
+  return SantaCacheHasher<uint64_t>(s.fsid) ^ (SantaCacheHasher<uint64_t>(s.fileid) << 1);
```

Comment:
Given the current hash function and that almost all the variation is in s.fileid, maybe it would be better to use (s.fsid << 1) ^ s.fileid instead.  Because otherwise, if there's only one s.fsid in use, the last bit on the output of this function will always be the last bit of s.fsid, and this will trickle down to SantaCache's hash function where the modulo by a power of 2 will preserve that last bit, and so we'll end up using only half of the available buckets, either all the even indices or all the odd indices, depending on the parity of s.fsid.

Reply:
Good call, that does help with distribution.